internal/cmd/egctl/accesslogs.go

```diff
@@ -0,0 +1,141 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package egctl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os/signal"
+	"strconv"
+	"syscall"
+
+	accesslogv3 "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v3"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+const (
+	// accessLogServerAddress is the listening address of the access-log server.
+	accessLogServerAddress = "0.0.0.0"
+	// accessLogServerPort is the default listening port of the access-log server.
+	accessLogServerPort = 9001
+)
+
+func NewAccessLogsCommand() *cobra.Command {
+	var (
+		port      int
+		gateway   string
+		listener  string
+		namespace string
+	)
+
+	accessLogsCommand := &cobra.Command{
+		Use:     "access-logs",
+		Short:   "Streaming access logs from Envoy Proxy.",
+		Example: ``, // TODO(sh2): add example
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return fmt.Errorf("missing the name of one gateway")
+			}
+
+			ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+			defer stop()
+
+			gateway = args[0]
+			if err := accessLogs(ctx, port, gateway, listener, namespace); err != nil {
+				return err
+			}
+
+			<-ctx.Done()
+			return nil
+		},
+	}
+
+	accessLogsCommand.PersistentFlags().IntVarP(&port, "port", "p", accessLogServerPort, "Listening port of access-log server.")
+	accessLogsCommand.PersistentFlags().StringVarP(&listener, "listener", "l", "", "Listener name of the gateway.")
+	accessLogsCommand.PersistentFlags().StringVarP(&namespace, "namespace", "n", "default", "Namespace of the gateway.")
+
+	return accessLogsCommand
+}
+
+type accessLogServer struct {
+	marshaler jsonpb.Marshaler
+}
+
+var _ accesslogv3.AccessLogServiceServer = &accessLogServer{}
+
+func newAccessLogServer() accesslogv3.AccessLogServiceServer {
+	return &accessLogServer{}
+}
+
+func (a *accessLogServer) StreamAccessLogs(server accesslogv3.AccessLogService_StreamAccessLogsServer) error {
+	log.Println("Start streaming access logs from server")
+	for {
+		recv, err := server.Recv()
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		str, _ := a.marshaler.MarshalToString(recv)
+		log.Println(str) // TODO(sh2): prettify the json output
+	}
+}
+
+func serveAccessLogServer(ctx context.Context, server *grpc.Server, addr string) {
```

Comment:
im guessing you are running this in the client machine where `egctl` is being run
it may not be possible for to create a grpc conn between the server and envoy

maybe the easiest way is to scrape pod logs of envoy using `kubectl`

Reply:
the basic implementation of this cmd has complete, left some doc and unit test. PTAL

Reply:
hi @shawnh2 lets assume im running this cmd from my macbook into a cluster that is running in GKE.
if you are running this access logs server locally on your Mac, how can you expect the envoy running in GKE to connect back ? routing rules will not exist

Reply:
this cmd works if you are running this access logs server in the same place that runs the cluster (for example also run this cmd in GKE), but won't work if you are running this cmd from your local macbook (there's no routing rules for sure).

imo, this cmd should work in both scenario, but this PR only implement the first.